cmd/datamon/cmd/root.go

```diff
@@ -102,12 +102,25 @@ func initConfig() {
 	handleConfigErrors(viper.ReadInConfig())
 
 	// 4. Initialize config and override via flags
-	config = new(CLIConfig)
-	if err := viper.Unmarshal(config); err != nil {
+	localConfig := new(CLIConfig)
+	if err := viper.Unmarshal(localConfig); err != nil {
 		wrapFatalln("config file contains invalid values", err)
-		return
+		return nil, err
 	}
 
+	return localConfig, nil
+}
+
+// initConfig reads in config file and ENV variables if set,
+// and sets config values based on file, env, cli flags.
+func initConfig() {
+	var err error
+	config, err = readConfig(os.Getenv(envConfigLocation))
+	if err != nil {
+		wrapFatalln("read config from file and env vars", err)
+	}
+
+	// ??? what errors follow if this block is removed?
```

Comment:
This is true that we should adopt a more thorough approach here and set credentials in all our calls to google apis explicitly from config (possibly this env var if no override is present) rather than letting the env spill down to the apis

Reply:
#391 suggests the opposite:  we always allow the sdk to try connecting and return errors as they occur from the sdk, wrapped with some additional info.

a further resolution that didn't occur to me when #391 was merged is that we could have a `logWarn` object or otherwise have a notion of warnings in the usual sense of something that might suggest an error but is not an error.  then we could warn upon calling the sdk with something that we know might fail but want to see if the sdk produces an error anyway.

with warnings functionality a somewhat separate piece, i'm going to consider this resolved by #391 .